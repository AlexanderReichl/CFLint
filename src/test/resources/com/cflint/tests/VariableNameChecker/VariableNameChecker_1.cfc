<!--- scoped use of name is detected, no unused arg flagged--->
<cfcomponent>
<cffunction name="helloWorld" access="public" returntype="string">
  <cfscript>
    var x = 1;
    ARGUMENTS.ALLcAPSWAYTOOLONGETETCETC;
    var temp = 1;
    var WayMuchTooWordyExample = '';
    var objBadName = '';
  </cfscript>
</cffunction>
</cfcomponent>