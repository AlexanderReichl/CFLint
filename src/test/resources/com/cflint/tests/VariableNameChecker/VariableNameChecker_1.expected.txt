[ {
  "severity" : "INFO",
  "id" : "SCOPE_ALLCAPS_NAME",
  "message" : "SCOPE_ALLCAPS_NAME",
  "category" : "CFLINT",
  "abbrev" : "SA",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "6",
    "message" : "Scope ARGUMENTS should not be upper case.",
    "variable" : "ARGUMENTS",
    "expression" : "ARGUMENTS.ALLcAPSWAYTOOLONGETETCETC"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_TOO_WORDY",
  "message" : "VAR_TOO_WORDY",
  "category" : "CFLINT",
  "abbrev" : "VT",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "8",
    "message" : "Variable WayMuchTooWordyExample is too wordy, can you think of a more concise name?",
    "variable" : "WayMuchTooWordyExample",
    "expression" : "var WayMuchTooWordyExample = ''"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_TOO_SHORT",
  "message" : "VAR_TOO_SHORT",
  "category" : "CFLINT",
  "abbrev" : "VT",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "5",
    "message" : "Variable x should be longer than 3 characters.",
    "variable" : "x",
    "expression" : "var x = 1"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_INVALID_NAME",
  "message" : "VAR_INVALID_NAME",
  "category" : "CFLINT",
  "abbrev" : "VI",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "6",
    "message" : "Variable ALLcAPSWAYTOOLONGETETCETC is not a valid name. Please use CamelCase or underscores.",
    "variable" : "ALLcAPSWAYTOOLONGETETCETC",
    "expression" : "ARGUMENTS.ALLcAPSWAYTOOLONGETETCETC"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_INVALID_NAME",
  "message" : "VAR_INVALID_NAME",
  "category" : "CFLINT",
  "abbrev" : "VI",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "8",
    "message" : "Variable WayMuchTooWordyExample is not a valid name. Please use CamelCase or underscores.",
    "variable" : "WayMuchTooWordyExample",
    "expression" : "var WayMuchTooWordyExample = ''"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_TOO_LONG",
  "message" : "VAR_TOO_LONG",
  "category" : "CFLINT",
  "abbrev" : "VT",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "6",
    "message" : "Variable ALLcAPSWAYTOOLONGETETCETC should be shorter than 20 characters.",
    "variable" : "ALLcAPSWAYTOOLONGETETCETC",
    "expression" : "ARGUMENTS.ALLcAPSWAYTOOLONGETETCETC"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_TOO_LONG",
  "message" : "VAR_TOO_LONG",
  "category" : "CFLINT",
  "abbrev" : "VT",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "8",
    "message" : "Variable WayMuchTooWordyExample should be shorter than 20 characters.",
    "variable" : "WayMuchTooWordyExample",
    "expression" : "var WayMuchTooWordyExample = ''"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_HAS_PREFIX_OR_POSTFIX",
  "message" : "VAR_HAS_PREFIX_OR_POSTFIX",
  "category" : "CFLINT",
  "abbrev" : "VH",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "9",
    "message" : "Variable has prefix or postfix objBadName and could be named better.",
    "variable" : "objBadName",
    "expression" : "var objBadName = ''"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_IS_TEMPORARY",
  "message" : "VAR_IS_TEMPORARY",
  "category" : "CFLINT",
  "abbrev" : "VI",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "7",
    "message" : "Temporary variable temp could be named better.",
    "variable" : "temp",
    "expression" : "var temp = 1"
  } ]
}, {
  "severity" : "INFO",
  "id" : "VAR_IS_TEMPORARY",
  "message" : "VAR_IS_TEMPORARY",
  "category" : "CFLINT",
  "abbrev" : "VI",
  "locations" : [ {
    "file" : "src/test/resources/com/cflint/tests/VariableNameChecker/VariableNameChecker_1.cfc",
    "fileName" : "VariableNameChecker_1.cfc",
    "function" : "helloWorld",
    "column" : "7",
    "line" : "9",
    "message" : "Temporary variable objBadName could be named better.",
    "variable" : "objBadName",
    "expression" : "var objBadName = ''"
  } ]
} ]